docs/usage/settings.md

```diff
@@ -128,6 +128,50 @@ Because python-dotenv is used to parse the file, bash-like semantics such as `ex
 (depending on your OS and environment) may allow your dotenv file to also be used with `source`,
 see [python-dotenv's documentation](https://saurabh-kumar.com/python-dotenv/#usages) for more details.
 
+## Secret Support
+
+Placing secret values in files is a common pattern to provide sensitive configuration to an application.
+
+A secret file follows the same principal as a dotenv file except it only contains a single value and the file name 
+is used as the key. A secret file will look like the following:
+
+/var/run/db_password
+```bash
+super_secret_database_password
+```
+
+Once you have your secret files, *pydantic* supports loading it in two ways:
+
+**1.** setting `secrets_dir` on `Config` in a `BaseSettings` class to the directory where your secret files are stored:
+
+```py
+class Settings(BaseSettings):
+    ...
+    database_password: SecretStr
+
+    class Config:
+        secrets_dir = '/var/run'
+```
+
+!!! note
+    Secrets are only loaded for fields defined as `SecretStr` or `SecretBytes`.
+
+**2.** instantiating a `BaseSettings` derived class with the `_secrets_dir` keyword argument:
+
+```py
+settings = Settings(_secrets_dir='/var/run')
+```
+
+In either case, the value of the passed argument can be any valid directory, either absolute or relative to the
+current working directory. From there, *pydantic* will handle everything for you by loading in your variables and
+validating them.
+
+Even when using a secrets directory, *pydantic* will still read environment variables from a dotenv file or the environment,
+**a dotenv file and environment variables will always take priority over values loaded from the secrets directory**.
+
+Passing a file path via the `_secrets_dir` keyword argument on instantiation (method 2) will override
+the value (if any) set on the `Config` class.
```

Comment:
Does this work with docker secrets and AWS Parameter Store, if so, please can we add a section to the docs saying  that and giving an example of how you'd use them.

Reply:
I don't have any experience with AWS so I can't speak to it. However, Docker secrets as far as I'm aware simply mount the secrets as volumes to /run/secrets so yes this should handle that use case.